atomic_reactor/plugins/pre_add_labels_in_df.py

```diff
@@ -194,64 +194,35 @@ def add_as_an_alias(set_to, set_from):
             self.log.debug("applied only some aliases, following old labels were not found: %s",
                            ", ".join(not_applied))
 
-        def check_if_all_same(labels_to_check, source):
-            """
-            checks if all specified equal labels have same values
-            within same scope (base or new_labels)
-            """
-            if labels_to_check:
-                list_to_check = list(labels_to_check)
-
-                for equal_name in list_to_check[1:]:
-                    if source[list_to_check[0]] != source[equal_name]:
-                        return False
-            return True
-
-        def set_missing_labels(labels_set, all_labels, value_from, not_in=(), not_value=None):
-            labels_to_set = all_labels.difference(labels_set)
-            list_labels_set = list(labels_set)
+        def set_missing_labels(labels_found, all_labels, value_from, not_in=[], not_value=None):
+            labels_to_set = all_labels.difference(set(labels_found))
 
             for set_label in labels_to_set:
-                if set_label in not_in and value_from[list_labels_set[0]] == not_value[set_label]:
+                if set_label in not_in and value_from[labels_found[0]] == not_value[set_label]:
                     self.log.debug("skipping label %r because it is set correctly in base image",
                                    set_label)
                 else:
-                    self.labels[set_label] = value_from[list_labels_set[0]]
+                    self.labels[set_label] = value_from[labels_found[0]]
                     self.log.warning("adding equal label %r with value %r",
-                                     set_label, value_from[list_labels_set[0]])
+                                     set_label, value_from[labels_found[0]])
 
-        fail_build = False
         for equal_list in self.equal_labels:
             all_equal = set(equal_list)
-            found_labels_base = set()
```

Comment:
because in set_missing_labels it is converted to set for comparison, but as list still can be used for indexing